src/Libraries/RevitNodes/Elements/FaceWall.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            InternalSetFaceWall(wall);
+        }
+
+        /// <summary>
+        /// Initialize a Wall element
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="wallType"></param>
+        /// <param name="reference"></param>
+        private void InitFaceWall(WallLocationLine location, Autodesk.Revit.DB.WallType wallType, Autodesk.Revit.DB.Reference reference)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+
+            
+            var wallElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.FaceWall>(Document);
+
+            // there is no way of comparing the existing element to the new reference.
+            // that's why the wall will be recreated.
+
+            var wall = Autodesk.Revit.DB.FaceWall.Create(Document, wallType.Id, location, reference);
+
+            InternalSetFaceWall(wall);
+            
+            TransactionManager.Instance.TransactionTaskDone();
+
+            ElementBinder.CleanupAndSetElementForTrace(Document, InternalFaceWall);
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the internal Element, ElementId, and UniqueId
+        /// </summary>
+        /// <param name="wall"></param>
+        private void InternalSetFaceWall(Autodesk.Revit.DB.FaceWall wall)
+        {
+            InternalFaceWall = wall;
+            InternalElementId = wall.Id;
+            InternalUniqueId = wall.UniqueId;
+        }
+
+        #endregion
+
+        #region Public static constructors
+        
+        /// <summary>
+        /// Create Wall By Face
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="wallType"></param>
+        /// <param name="surface"></param>
+        /// <returns></returns>
+        /// Note: This moved to the Wall element
+        [Autodesk.DesignScript.Runtime.IsVisibleInDynamoLibrary(false)]
+        public static FaceWall ByFace(WallLocationLine location, WallType wallType, Autodesk.DesignScript.Geometry.Surface surface)
+        {
+            object reference = surface.Tags.LookupTag("RevitFaceReference");
+            if (reference != null && reference.GetType() == typeof(Reference))
+            {
+                try
+                {
+                    Reference refr = reference as Reference;
+                    return new FaceWall(location, wallType.InternalWallType, refr);
+                }
+                catch (Exception) { throw new Exception(Properties.Resources.InvalidFace); }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Create Wall By Face
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="wallType"></param>
+        /// <param name="reference"></param>
+        /// <returns></returns>
+        [Autodesk.DesignScript.Runtime.IsVisibleInDynamoLibrary(false)]
```

Comment:
Yes, Initially I added FaceWall constructors to the FaceWall class since they are of Type FaceWall and not Wall. But for clarity @kronz , @mjkkirschner and me decided to add it to walls, which makes sense. I kept the old constructors in this file and just hid them. I'll remove them now.